src/foam/dashboard/view/GroupedDAOTableWidget.js

```diff
@@ -32,7 +38,13 @@ foam.CLASS({
     async function init() {
       this.SUPER();
       var self = this;
-      result = await self.__subContext__[self.daoKey].where(self.predicate).select(self.GroupBy.create({
+
+      var coTxnPredicate = self.AND(
```

Comment:
This should be in a jrl, as a config-able predicate. Otherwise, this would make all groupedDAOTables use this pred

Reply:
@sarthak-marwaha yes, you are right.  I know what  you mean. I tried with this pred in jrl(see below code). somehow `citationView` never gets hit. I also tried other menu with the `citationView`. This view never gets displayed. 


`p({
  "class": "foam.nanos.menu.Menu",
  "id": "cards.dashboard.widgets.monthlyspending",
  "label": "Monthly Transactions",
  "handler": {
    "class": "foam.nanos.menu.ViewMenu",
    "view": {
      "class": "foam.dashboard.view.CardWrapper",
      "mode": "config",
      "currentView": {
        "class": "net.nanopay.tx.TransactionsGroupingWidget",
        "daoKey": "transactionDAO",
        "predicate":{
          "class": "foam.mlang.predicate.IsInstanceOf",
          "targetClass": "net.nanopay.tx.cico.COTransaction"
        },
        "citationView": "foam.dashboard.view.CenteredDashboardCitationView",
        "arg1":  { "class": "foam.mlang.expr.DateGroupingExpr" },
        "arg2": {
          "class": "foam.mlang.sink.Sum",
          "arg1": {
            "class": "__Property__",
            "forClass_": "net.nanopay.tx.model.Transaction",
            "name": "amount"
          }
        },
        "viewMore": true,
        "emptyTitle": "No Transactions!",
        "emptySubTitle": "Please add an account"
      }
    }
  },
  "parent": "hidden"
})
`